pkg/azureutils/informer_common.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azureutils
+
+import (
+	"context"
+
+	crdInformers "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	azdiskinformers "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/informers/externalversions"
+)
+
+type InformerFactory interface {
+	Start(stopCh <-chan struct{})
+	Informer() cache.SharedIndexInformer
+}
+
+type azNodeInformer struct {
```

Comment:
We since `azNodeInformer`, `azVolumeAttachmentInformer` and `azVolumeInformer` essentially have the same type definition and the same implementation of the interface, we should define one type, say `GenericInformerInfo` and implement the interface for that type. We will later initialize `azNodeInformer`, `azVolumeAttachmentInformer` and `azVolumeInformer` as `GenericInformerInfo`.

Reply:
You're right, there's no need to distinct them.